docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_global_services_enabled_remediation.mdx

```diff
@@ -1,6 +1,219 @@
+
+4. **Verify Configuration**:
+   - Check the trail status to ensure it is logging events.
+
+This script ensures that your CloudTrail is configured to record both regional and global events, preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the CloudTrail console at https://console.aws.amazon.com/cloudtrail/.
+
+2. In the navigation pane, click on "Trails". 
+
+3. In the Trails page, select the trail you want to check. 
+
+4. In the details pane, under "S3", check the "Log file SSE-KMS encryption" field. If it is set to "Yes", then the trail is recording both regional and global events. If it is set to "No", then the trail is not recording global events.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access CloudTrail.
+
+2. Once AWS CLI is installed and configured, you can use the following command to list all the trails:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+   This command will return a JSON output with details of all the trails.
+
+3. To check if a trail is recording both regional and global events, you need to look at the 'IncludeGlobalServiceEvents' and 'IsMultiRegionTrail' fields in the output. If 'IncludeGlobalServiceEvents' is true, it means the trail is recording global events. If 'IsMultiRegionTrail' is true, it means the trail is recording events from all regions.
+
+4. You can use the following command to filter the output and check these fields for all trails:
+
+   ```
+   aws cloudtrail describe-trails --query 'trailList[*].[Name,IncludeGlobalServiceEvents,IsMultiRegionTrail]'
+   ```
+   This command will return a list of all trails with their names and the values of 'IncludeGlobalServiceEvents' and 'IsMultiRegionTrail'. If any trail has 'false' for either of these fields, it means it is not recording both regional and global events.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing Python scripts to interact with AWS, you need to install the necessary libraries. The primary library you need is boto3, which is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python. You can install it using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Set up AWS credentials: Boto3 needs your AWS credentials (access key and secret access key) to interact with AWS services. You can set them in your environment variables:
+
+   ```python
+   import os
+   os.environ['AWS_ACCESS_KEY_ID'] = 'your_access_key'
+   os.environ['AWS_SECRET_ACCESS_KEY'] = 'your_secret_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>